src/app/patient/patient.cpp

```diff
@@ -66,3 +66,36 @@ void Patient::createPatient(const std::string& name, const std::string& surname,
     Patient patient{name, surname, pesel, allergens};
     Clinic::appendPatient(std::make_shared<Patient>(patient));
 }
+
+std::shared_ptr<Patient>& Patient::getPatient(const std::string& name, const std::string& surname)
+{
+    auto patients = Clinic::getPatients();
+    auto exact_patient = std::find_if(patients.begin(), patients.end(), [&](const auto& patient) {
+        return (patient->getName() == name) && (patient->getSurname() == surname);
+    });
+
+    if (exact_patient != patients.end())
+    {
+        return *exact_patient;
+    }
+    else
+    {
+        throw std::runtime_error("patient not found.");
+    }
+}
```

Comment:
What if we'll find few patients with the same name and surname? Please fix this so it'll return some container with pointers to these objects.

Reply:
I am a bit concerned if we should provide more than one patient, maybe search function by name should be extended to search by name and birth date?
From my perspective if patient found by name and surname is not correct then find by pesel should be called.
It highly depends on how system integration with user will look like.
Imagine a receptionist asking patient which pesel matches his from received list of patients.

Reply:
Well, we can only be sure to return only one object when searching by UNIQUE attribute, so there can always be a combination of patients with exactly same attributes unless we clearly decline some cannot be duplicated(like pesel). For example receptionist will get a list of few patients with the same name, it is in her interest to validate which person is correct one, she will receive all information with this request and then get the right pointer to the object for further modification.